### DISCUSSION TITLE
How can I do 'Save Fields and Mesh' in HFSS Optimetrics using PyAEDT?

### DISCUSSION THREAD

**Discussion by Dicksky** [ansys/pyaedt#4663] [Q&A] [answered]
Hello:) I am using HPC with the Slurm system to run my HFSS simulations, sometimes after running HPC doesn't give me any results if the convergence criterion is not fulfilled, I want to see the result of the last iteration, will the 'Save Fields and Mesh' function solve the problem? And how can I do it using PyAEDT?

**Comment by mahesh-dhanekula** [comment]
@Dicksky From the description, I feel your question has got nothing to do with PyAEDT. You can reach out to Ansys Technical Support to determine what's going wrong in your simulations.
If you have any **How To** question on scripting certain functionality of AEDT, you can post here.

**Reply by wsp6** [reply]
His second sentence is a direct PyAEDT question: How do you enable 'Save Fields and Mesh' for a parametric setup through PyAEDT?

I have also run into this problem. By default, the 'Save Fields and Mesh' option is disabled. I need it enabled so I can run far field reports on the variations.  I'm using PyAEDT (specifically: Hfss.parametrics.add_from_file ) to pull in a CSV with my variations.

How do I enable 'Save Fields and Mesh' for a parametric setup through the PyAEDT API?

**Comment by wsp6** [comment]
I don't know if the 'Save Fields and Mesh' will solve your problem, but here's how I figured out how to set the option:

hfss.get_oo_name(aedt_object=hfss.ooptimetrics)    will give you the names of the parametric setups you have in your design

hfss.get_oo_properties(aedt_object=hfss.ooptimetrics,object_name='nameofparametricsetup') will give you the property names for PyAEDT, they often aren't the same as what is shown in the AEDT GUI.

hfss.get_oo_property_value(aedt_object=hfss.ooptimetrics, object_name='nameofparametricsetup', prop_name='SaveFields') gives you the current value of the 'Save Fields and Mesh' setting
 
hfss.set_oo_property_value(aedt_object=hfss.ooptimetrics, object_name='nameofparametricsetup', prop_name='SaveFields', value='True') will enable 'Save Fields and Mesh'

**Comment by wsp6** [comment, marked as answer]
I ended up using the following command to enable saving fields in a parametric setup:

hfss.set_oo_property_value(aedt_object=hfss.ooptimetrics, object_name="Parametric _Setup_Name", prop_name='SaveFields', value='True')

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by weblyang** [reply]
Thank you very much, this worked perfectly. Initially, it didn't work. I suspect it might be because I performed too many operations on the already opened HFSS project, causing the HFSS project to become a bit messy.  Later, I restarted the project, and then it worked exceptionally well. Thank you again, you've been a great help.
